README_zenith.md

```diff
@@ -0,0 +1,88 @@
+Components & modifications:
+
+- zenith_* tools
+- WAL streaming tools
+- smgr changes
+- WAL redo helper 
+- random hacks to make zenith work
+- debug
+
+
+zenith_* tools
+----------------
+
+#### zenith_s3
+Helpers for s3 auth and operations.
+
+#### zenith_push
+zenith_push is a standalone executable. It take a base image of a database and
+uploads it to S3 in the zenith storage format.
+
+Also used as archive_command, to upload raw WAL segments to S3, as they're
+filled up.
+This is pretty similar to WAL-G. Storage format in S3 is different.
+
+#### zenith_restore
+
+Another standalone executable. Downloads and restores a base image from S3.
+This only downloads the "non rel" image and WAL from S3. The relation data
+files are created as "lazy" files, so that they are restored on demand, when
+they're first accessed.
+
+#### zenith_slicedice
+
+Another standalone executable. Reads raw WAL from the archive in
+S3. Splits it per relation, and writes it to per-relation WAL files.
+This only operates on the files in the S3 bucket, it doesn't require access to
+the primary.
+
+
+WAL streaming tools
+-----------------
+
+#### safekeeper
+Proxy-safekeeper communication consensus protocol.
+See src/bin/safekeeper/README.md for details.
```

Comment:
I believe we can remove src/bin/safekeeper now, it has been superseded by src/backend/replication/walproposer.c. But let's keep/move the README that explains how it works, and check that it's up-to-date.